src/services/EdFi.DataManagementService.Api/Core/Exceptions/IDetailedException.cs

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json.Serialization;
+
+namespace EdFi.DataManagementService.Api.Core.Exceptions
+{
+    public interface IDetailedException
```

Comment:
I think these 6 files can all be collapsed into a single record. Basically just convert this `IDetailedException` into a record. It will be automatically JSON serializable, with simple syntax to `new()` the fields. Also, this models a failure body, so I'd just call the record `FailureBody`. `Exception` in the name is confusing because it sounds like something you throw.

Reply:
Actually, how about `FailureResponseBody`? Then the namespace could be `EdFi.DataManagementService.Api.Core.ResponseBody` for when we have additional standardized response body formats.

Reply:
@bradbanister 
Just to confirm: for any exception we don't throw it, but will wrap with front end response, is it correct ?

Reply:
@bradbanister 
Just seeing https://github.com/Ed-Fi-Alliance-OSS/Data-Management-Service/pull/16. I modified the code very similar to that with static methods to return specific error details. Please help review !